WalletWasabi.Gui/Controls/QrCode.cs

```diff
@@ -30,6 +32,42 @@ public QrCode()
 				.Where(x => x != null)
 				.ObserveOn(RxApp.MainThreadScheduler)
 				.Subscribe(x => FinalMatrix = AddPaddingToMatrix(x));
+
+			this.WhenAnyValue(x => x.QRImageSavePath)
+				.Where(x => !string.IsNullOrWhiteSpace(x) || !string.IsNullOrEmpty(x))
+				.Where(x => FinalMatrix != null)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(x => GenerateQRCodeBitmap(x));
+		}
+
+		private void GenerateQRCodeBitmap(string x)
+		{
+			try
+			{
+				var pixSize = PixelSize.FromSize(CoercedSize, 1);
+				using var rtb = new RenderTargetBitmap(pixSize);
+
+				rtb.Render(this);
+				rtb.Save(x);
```

Comment:
This is hard to follow. I would create something like a `QrCode.SaveToFile(path) `and call it from the `AddressViewModel.SaveQRCodeAsync()`.
With this you can also remove: `QrCode.QRImageSavePath`. IMO there is no need to store the Path after the save, so you could remove it from everywhere. Or maybe I missed the point there?